priv/templates/nimble_template/lib/otp_app_web/plugs/check_empty_body_params_plug.ex.eex

```diff
@@ -0,0 +1,23 @@
+defmodule <%= web_module %>.CheckEmptyBodyParamsPlug do
+  @behaviour Plug
+
+  import Phoenix.Controller, only: [put_view: 2, render: 3]
+  import Plug.Conn
+
+  alias <%= web_module %>.Api.ErrorView
+
+  @apply_for_methods ["POST", "PUT", "PATCH"]
+
+  def init(opts), do: opts
+
+  def call(%{body_params: body_params, method: method} = conn, _opts)
```

Comment:
Are we checking here that `post/put/patch` request body empty or not, if it is empty will render an error, is it?
 
If it then I am just wondering if the patch API is like `patch https://example.com/event/:id/cancel` where it will be just to cancel the event and in this case, there might not be needed no any request body params due to URL. So in this case our app will return an error.

I think the main question is about our API conventions, 
-  Should we make `request body` mandatory for every `post/put/patch` API endpoint
-  Should we make `request body` optional for every `post/put/patch and others` API endpoint

Reply:
Good point @junan I think for now we only validate the POST request only, the PATCH could have an empty body. What do you think @byhbt

Reply:
yup, good catch Junan, I agree with the solution we validate for the POST request only.

Reply:
I have updated it in https://github.com/nimblehq/elixir-templates/pull/164/commits/6a79d4d029614e6ed5d5e5c2b067b68f391898ad FYi @junan @byhbt, so we are checking only for the POST request